gcc/d/dfrontend/expression.c

```diff
@@ -12374,7 +12582,9 @@ Expression *AssignExp::semantic(Scope *sc)
 
     type = e1->type;
     assert(type);
-    return op == TOKassign ? reorderSettingAAElem(sc) : this;
+    Expression *result = op == TOKassign ? reorderSettingAAElem(sc) : this;
+    checkAssignEscape(sc, result, false);
+    return result;
```

Comment:
Nope, there's something funny going on here that triggers the error.  The build error infact shouldn't happen because attributes *should* be inferred.